File: backend/app/models/feedback.py
from __future__ import annotations
import uuid
from datetime import datetime
from typing import Optional

from sqlalchemy import DateTime, String, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Feedback(Base):
    __tablename__ = "feedback"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid())
    user_email: Mapped[str] = mapped_column(String(255), nullable=False)
    page_url: Mapped[Optional[str]] = mapped_column(String(512))
    category: Mapped[Optional[str]] = mapped_column(String(32))
    message: Mapped[str] = mapped_column(Text, nullable=False)
    resolved_at: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True))
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

File: frontend/src/api/feedback.ts
import client from "./client";

export interface FeedbackItem {
  id: string;
  user_email: string;
  page_url: string | null;
  category: string | null;
  message: string;
  resolved_at: string | null;
  created_at: string;
}

export interface FeedbackSubmit {
  message: string;
  page_url?: string | null;
  category?: string | null;
}

export const submitFeedback = (data: FeedbackSubmit) =>
  client.post<FeedbackItem>("/feedback", data).then((r) => r.data);

export const fetchFeedback = () =>
  client.get<FeedbackItem[]>("/feedback").then((r) => r.data);

export const toggleFeedbackResolved = (id: string) =>
  client.post<FeedbackItem>(`/feedback/${id}/resolve`).then((r) => r.data);
